### PR TITLE
Feature/problem statement

### PR DESCRIPTION
**Pull request by Rutu2k**
Added Problem Statement Component , with API integration (assuming dummy API) , added redux store and redux saga , Done with unit testing of ProblemStatementReducer.
Please review and suggest changes if any.

### CODE REVIEW

**Review by Suhas010 — APPROVED** [review]

File: src/apis/problemStatementAPI.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import axios from 'axios';
+import { SERVER_URL } from 'constants/appConstants';
```
Comment (vaibhavchalse99) [inline src/apis/problemStatementAPI.js:2]: Add blank line before this line

File: src/components/ProblemsComponent/index.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Container } from 'core-components';
+import PropTypes from 'prop-types';
```
Comment (vaibhavchalse99) [inline src/components/ProblemsComponent/index.jsx:2]: follow the import sequence
prop-types is a third party module import it 1st

File: src/containers/ProblemsContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import ProblemsComponent from 'components/ProblemsComponent';
+import { statementRequest } from '../actions/problemStatementActions';
```
Comment (vaibhavchalse99) [inline src/containers/ProblemsContainer.jsx:5]: use absolute path

File: src/reducers/rootReducer.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { combineReducers } from 'redux';
+import ProblemStatementReducer from 'reducers/ProblemStatementReducer';
```
Comment (vaibhavchalse99) [inline src/reducers/rootReducer.js:2]: add blank space before it

File: src/sagas/problemsSaga.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { call, put, takeLatest } from 'redux-saga/effects';
+
+import { getStatement } from 'apis/problemStatementAPI';
+import { PROBLEM_STATEMENT } from 'constants/actionConstants';
```
Comment (vaibhavchalse99) [inline src/sagas/problemsSaga.js:4]: constants must be imported at the end

File: src/sagas/problemsSaga.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import { statementAction, statementActionFailed } from 'actions/problemStatementActions';
+
+// Problem statememt Saga
+export function* statementSaga() {
+  try {
+    const response = yield call(getStatement);
```
Comment (vaibhavchalse99) [inline src/sagas/problemsSaga.js:10]: you can use destructuring if you need only data key
const {data} = yield call(getStatement)

File: src/sagas/rootSagas.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import { all } from 'redux-saga/effects';
+import statementRequestSaga from './problemsSaga';
```
Comment (vaibhavchalse99) [inline src/sagas/rootSagas.js:2]: add blank space and use absolute path

File: src/apis/problemStatementAPI.js (hunk 1 above)
Comment (vaibhavchalse99) [reply src/apis/problemStatementAPI.js:3]: why filename is problemStatementAPI.js
follow camelcase strictly
it should be problemStatementApi.js

File: src/components/ProblemsComponent/index.jsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+      <h6 className='text-white font-weight-light'>{data}</h6>
+    </Container>
+  );
+};
+
+ProblemComponent.propTypes = {
+  data: PropTypes.string.isRequired,
+};
+
+export default ProblemComponent;
```
Comment (shailendrakanherkar18) [inline src/components/ProblemsComponent/index.jsx:21]: memo

File: src/containers/ProblemsContainer.jsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  return (
+    flag
+      ? <ProblemsComponent data={requestError} />
+      : <ProblemsComponent data={data} />
+  );
+};
+
+export default ProblemsContainer;
```
Comment (shailendrakanherkar18) [inline src/containers/ProblemsContainer.jsx:30]: memo

**Review by shailendrakanherkar18 — DISMISSED** [review]

**Pull request merged**